var/spack/repos/builtin/packages/libxsmm/package.py

```diff
@@ -60,22 +60,42 @@ class Libxsmm(Package):
             description='Unoptimized with call-trace (LIBXSMM_TRACE).')
     variant('header-only', default=False,
             description='Produce header-only installation')
+    
+    def make_args(self):
+        options = [
+            "CC=%s" % spack_cc,
+            "CXX=%s" % spack_cxx,
+            "FC=%s" % spack_fc,
+            'SYM=1' ] # include symbols by default
 
-    def patch(self):
-        kwargs = {'ignore_absent': False, 'backup': False, 'string': True}
-        makefile = FileFilter('Makefile.inc')
+        # JIT (AVX and later) makes MNK, M, N, or K spec. superfluous
+#       make_args += ['MNK=1 4 5 6 8 9 13 16 17 22 23 24 26 32']
+
+        # include call trace as the build is already de-optimized
+        if '+debug' in self.spec:
+            options += ['DBG=1']
+            options += ['TRACE=1']
 
-        # Spack sets CC, CXX, and FC to point to the compiler wrappers
-        # Don't let Makefile.inc overwrite these
-        makefile.filter('CC = icc',         'CC ?= icc', **kwargs)
-        makefile.filter('CC = gcc',         'CC ?= gcc', **kwargs)
-        makefile.filter('CXX = icpc',       'CXX ?= icpc', **kwargs)
-        makefile.filter('CXX = g++',        'CXX ?= g++', **kwargs)
-        makefile.filter('FC = ifort',       'FC ?= ifort', **kwargs)
-        makefile.filter('FC = gfortran',    'FC ?= gfortran', **kwargs)
+        return options
 
-    def manual_install(self, prefix):
+    @run_before('build')
+    def header_check(self):
         spec = self.spec
+        if '+header-only' in spec and '@1.6.2:' not in spec:
+            raise RuntimeError(
```

Comment:
@adamjstewart, I am not sure, but would `conflict('+header-only',when='@1.6.2:')` do the same?